src/codeu/chat/client/commandline/Chat.java

```diff
@@ -99,7 +112,7 @@ private Panel createRootPanel(final Context context) {
     //
     panel.register("help", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
+      public void invoke(List<String> args) {
```

Comment:
It required changing a bunch of code, but I wanted to commend these changes to plumb the `List<String> args` around instead of passing around a `Scanner` object. 

Where possible, it's nicer to pass around immutable data (including strings) than complex objects that actually do things. The resulting code is easier to test, because you don't need to build mock or fake versions of your complex objects in order to try calling any of your other functions from a test, and it's harder for changing something in one place to have unintended side effects elsewhere (which could totally happen if somehow two threads ended up sharing a `Scanner`).